src/org/gk/qualityCheck/QACheckUtilities.java

```diff
@@ -26,4 +33,297 @@ public static GKInstance getLatestCuratorIEFromInstance(GKInstance instance) thr
+        }
+        return false;
+    }
+
+    // Returns true if 'species' is a valid attribute in PhysicalEntity.
+    public static boolean hasSpeciesAttribute(GKInstance physicalEntity) {
+        return physicalEntity.getSchemClass().isValidAttribute(ReactomeJavaConstants.species);
+    }
+
+    /**
+     * Finds all parent DbIds of the incoming Event, and then checks if any of them are in the skiplist of Pathway DbIds.
+     * @param event GKInstance -- Event that is being checked for membership in a skiplist Pathway.
+     * @return boolean -- true if member of skiplist Pathway, false if not.
+     * @throws Exception -- Thrown by MySQLAdaptor.
+     */
+    public static boolean memberSkipListPathway(GKInstance event) throws Exception {
+
+        if (skiplistDbIDs != null) {
+            // Finds all parent Event DbIds.
+            Set<String> hierarchyDbIds = findEventHierarchyDbIds(event, new HashSet<>());
+            // Check if any returned Event DbIds (including original Events) are in skiplist.
+            for (String skiplistDbId : skiplistDbIDs) {
+                if (hierarchyDbIds.contains(skiplistDbId)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Finds parent DbIds of incoming Event through hasEvent referrers. Recurses until there are no more referrers.
+     * @param event GKInstance -- Event that is being checked for referrers. Its DbId is added to the Set being built.
+     * @param dbIds Set<String> -- Current Set of parent DbIds of Event.
+     * @return Set<String> -- Once TopLevelPathway has been found, returns all DbIds, inclusive, between TopLevelPathway and original Event.
+     * @throws Exception -- Thrown by MySQLAdaptor.
+     */
+    private static Set<String> findEventHierarchyDbIds(GKInstance event, Set<String> dbIds) throws Exception {
+        dbIds.add(event.getDBID().toString());
+        Collection<GKInstance> hasEventReferrals = event.getReferers(ReactomeJavaConstants.hasEvent);
+        if (hasEventReferrals != null) {
+            for (GKInstance hasEventReferral : hasEventReferrals) {
+                dbIds.addAll(findEventHierarchyDbIds(hasEventReferral, dbIds));
+            }
+        }
+        return dbIds;
+    }
+
+    public static void setSkipList(List<String> skippedDbIds) {
+        skiplistDbIDs = skippedDbIds;
+    }
+
+    public static void setHumanSpeciesInst(MySQLAdaptor dba) throws Exception {
+        humanSpeciesInst = dba.fetchInstance(48887L);
+    }
+
+    // Sometimes created or species instances are not populated. This helper method checks if attribute value is
+    // populated. If it isn't, returns null.
+    public static String getInstanceAttributeName(GKInstance instance, String attribute) throws Exception {
+        GKInstance attributeInstance = (GKInstance) instance.getAttributeValue(attribute);
+        return attributeInstance != null ? attributeInstance.getDisplayName() : null;
```

Comment:
Funnily enough, this particular example is actually a case where the returned value is going to appear in the output report, and I wanted it to have 'null' so that it was visible. I'll keep this (or maybe returning 'N/A' makes more sense?) in this specific case, but I'll look through the code to make sure other cases like this return empty values

Reply:
I am going to change this method's name so that it ends with "ForOutputReport"

Reply:
Yeah, I still think it makes sense to have a String object returned (empty or "N/A").  If you need it as a marker, a String describing what it is marking is preferrable to a `null` value in code.

Avoiding returning null values has become popular enough that there is actually a pattern to avoid it: https://en.wikipedia.org/wiki/Null_object_pattern#Java